com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/topology.py

```diff
@@ -1285,6 +1285,47 @@ def trigger(self, when=1):
             raise ValueError(when)
         return tw
 
+    def aggregate(self, function, name=None, schema = None):
+        """Declares a function or callable to aggregate the contents of 
+        the window when it is triggered.
+        
+        The supplied function is passed a list containing the contents 
+        of the window. The return values of the function are passed as
+        the tuples on the returned `stream`. For example, a window that
+        calculates a moving average of the last  10 tuples could be 
+        written as follows:  
+        
+            win = s.last(10).trigger(1)
+            moving_averages = win.aggregate(lambda tuples: sum(tuples)/len(tuples))
+            
+        Args:
+            function: The function which aggregates the contents of the window
+                
+        Returns: 
+            Stream: A `Stream` of the returned values of the supplied function.                                                                                                                                                             
+        """
+        # WIP: TODO: support other window types and policies                                                                                                                                                                               
+
+        if self._config['evictPolicy'] != 'COUNT' or self._config['triggerPolicy'] != 'COUNT':
+            raise NotImplementedError("Currently, only windows with eviction policies of type COUNT and trigger policies of type COUNT are supported")
+
+        if schema is None:
+            schema = self.stream.oport.schema
+        
+        sl = _SourceLocation(_source_info(), "window")
```

Comment:
The action here is 'aggregate', not 'window'.

Reply:
Fixed.